### ISSUE TITLE
Specify client protocol in config

### ISSUE THREAD

**Issue by mojodna** [terranodo/tegola#134]
When running behind a reverse proxy, SSL may be available, so HTTPS URLs should be able to be generated automatically.

**Comment by ARolek** [comment]
tegola [sniffs for TLS](https://github.com/terranodo/tegola/blob/master/server/handle_map_capabilities.go#L46) on the capabilities endpoint. Are you saying that you have TLS between the proxy and the client, but not tegola and the proxy and therefore need a way to "force" https urls in the capabilities endpoint?

**Comment by mojodna** [comment]
Correct. `X-Forwarded-Proto` is one way (implemented in #135), but an alternate method would be useful if that header isn't being passed.
